dbms/src/IO/WriteBufferFromTemporaryFile.cpp

```diff
@@ -0,0 +1,88 @@
+#include <DB/IO/WriteBufferFromTemporaryFile.h>
+
+#include <Poco/File.h>
+#include <Poco/Path.h>
+
+#include <DB/IO/ReadBufferFromFile.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+	extern const int CANNOT_OPEN_FILE;
+	extern const int CANNOT_SEEK_THROUGH_FILE;
+}
+
+
+WriteBufferFromTemporaryFile::Ptr WriteBufferFromTemporaryFile::create(const std::string & path_template_)
+{
+	std::string path_template = path_template_;
+
+	if (path_template.empty() || path_template.back() != 'X')
+		path_template += "XXXXXX";
+
+	Poco::File(Poco::Path(path_template).makeParent()).createDirectories();
+
+	int fd = mkstemp(const_cast<char *>(path_template.c_str()));
```

Comment:
Ok, it make sense to make codebase uniform.
However `Poco::TemporaryFile` doesn't allow to create tmp files with custom template like `/tmp/dir/user_name.??????` and provides only application-level guarantees about an uniqueness and an atomicity of file creation.